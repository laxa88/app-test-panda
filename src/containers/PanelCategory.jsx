import React from "react";

const PanelCategory = props => <div>Category panel</div>;

export default PanelCategory;
