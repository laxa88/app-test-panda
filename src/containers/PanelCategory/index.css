.body {
  display: flex;
  height: 110px;
}

.left {
  display: flex;
}

.right {
  display: flex;
}
