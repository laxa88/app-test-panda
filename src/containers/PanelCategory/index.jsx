import React from 'react';

import css from './index.css';
import ItemCategory from '../../components/ItemCategory';
import { ThemeContext } from '../../theme';

class PanelCategory extends React.PureComponent {
  render() {
    const { theme } = this.context;

    return (
      <div className={css.body}>
        <div className={css.left}>
          <ItemCategory
            faIcon={['fas', 'puzzle-piece']}
            text1="Sector"
            text2="Sports"
          />

          <ItemCategory
            faIcon={['far', 'futbol']}
            text1="Sport Type"
            text2="Bicycles"
          />

          <ItemCategory
            faIcon={['fas', 'bicycle']}
            text1="Mode"
            text2="Mountain Cross"
          />
        </div>

        <div className={css.right}>
          <ThemeContext.Provider
            value={{
              theme: {
                ...theme,
                quaternaryColor: theme.primaryColor,
              },
            }}
          >
            <ItemCategory
              faIcon={['fas', 'recycle']}
              text1="Responsibility"
              text2="Go Green!"
            />
          </ThemeContext.Provider>
        </div>
      </div>
    );
  }
}

PanelCategory.contextType = ThemeContext;

export default PanelCategory;
