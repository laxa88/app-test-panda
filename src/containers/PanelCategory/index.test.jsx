import React from 'react';
import renderer from 'react-test-renderer';

import PanelCategory from './index';

describe('<PanelCategory />', () => {
  it('renders', () => {
    const result = renderer.create(<PanelCategory />);
    expect(result).toMatchSnapshot();
  });
});
