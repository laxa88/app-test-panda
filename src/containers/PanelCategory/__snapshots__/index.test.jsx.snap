// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PanelCategory /> renders 1`] = `
<div
  className="body"
  style={
    Object {
      "backgroundColor": "#F0F0F0",
      "borderTop": "1px solid #DDDDDD",
    }
  }
>
  <div
    className="left"
  >
    <div
      className="body"
    >
      Mocked_FontAwesomeIcon (fas,puzzle-piece)
      <div
        className="text1"
        style={
          Object {
            "color": "#A4A4A4",
          }
        }
      >
        Sector
      </div>
      <div
        className="text2"
        style={
          Object {
            "color": "#333333",
          }
        }
      >
        Sports
      </div>
    </div>
    <div
      className="body"
    >
      Mocked_FontAwesomeIcon (far,futbol)
      <div
        className="text1"
        style={
          Object {
            "color": "#A4A4A4",
          }
        }
      >
        Sport Type
      </div>
      <div
        className="text2"
        style={
          Object {
            "color": "#333333",
          }
        }
      >
        Bicycles
      </div>
    </div>
    <div
      className="body"
    >
      Mocked_FontAwesomeIcon (fas,bicycle)
      <div
        className="text1"
        style={
          Object {
            "color": "#A4A4A4",
          }
        }
      >
        Mode
      </div>
      <div
        className="text2"
        style={
          Object {
            "color": "#333333",
          }
        }
      >
        Mountain Cross
      </div>
    </div>
  </div>
  <div
    className="right"
  >
    <div
      className="body"
    >
      Mocked_FontAwesomeIcon (fas,recycle)
      <div
        className="text1"
        style={
          Object {
            "color": "#A4A4A4",
          }
        }
      >
        Responsibility
      </div>
      <div
        className="text2"
        style={
          Object {
            "color": "#33A00E",
          }
        }
      >
        Go Green!
      </div>
    </div>
  </div>
</div>
`;
